Jenkinsfile

```diff
@@ -0,0 +1,85 @@
+pipeline {
+  agent { label 'subman' }
+  stages {
+    // stage('prepare') {steps {echo 'prepare'}}
+    stage('Test') {
+      parallel {
+        stage('stylish') {
+          agent { label 'subman-centos7' }
+          steps { sh readFile(file: 'jenkins/stylish-tests.sh') }
+        }
+        stage('tito') {
+          agent { label 'rpmbuild' }
+          steps { sh readFile(file: 'jenkins/tito-tests.sh') }
+        }
+        stage('RHEL7 unit') {
+          agent { label 'subman-centos7' }
+          steps {
+            // echo "skipping for debug..."
+            sh readFile(file: 'jenkins/nose-tests.sh')
+            junit('nosetests.xml')
+            // publishCoverage('coverage.xml')
+            }
+        }
+        // TODO: figure if this is needed and implement
+        // stage('RHEL8 unit') {steps {echo 'nose'}}
+        stage('Fedora unit') {
+          steps {
+            sh readFile(file: 'jenkins/python3-tests.sh')
+            junit('nosetests.xml')
+            // TODO: find the correct adapter or generate coverage tests that can be
+            //       parsed by an existing adapter:
+            //       https://plugins.jenkins.io/code-coverage-api/
+            // publishCoverage adapters: [jacocoAdapter('coverage.xml')]
+          }
+        }
+        stage('opensuse42') {
+          agent { label 'opensuse42' }
+          steps { sh readFile(file: 'jenkins/suse-tests.sh') }
+        }
+        // stage('sles11') {
+        //   // FIXME:  sles11 can't be tested due missing python deps (incl. nose)
+        // }
+        stage('sles12') {
+          agent { label 'sles12' }
+          steps { sh readFile(file: 'jenkins/suse-tests.sh') }
+        }
+        // TODO: add after QE creates pipeline
+        // stage('Functional') {
+        //   stages{
+        //     stage('Build RPM') {steps {echo 'Build RPM'}}
+        //     stage('Prepare') {steps {echo 'Prepare'}}
+        //     stage('Provision') {steps {echo 'Provisioning'}}
+        //     stage('Tier 1') {steps {echo 'Tier 1'}}
+        //   }
+        // }
+      }
+    }
+    stage('SUSE Builds') {
+      matrix {
+        axes {
+          axis {
+            name 'PLATFORM'
+            values 'openSUSE_Leap_42.2', 'SLE_12_SP1', 'SLE_11_SP4'
+          }
+        }
+        stages {
+          stage('Build') {
+            agent { label 'opensuse42' }
+            steps {
+              sh "scripts/suse_build.sh 'home:kahowell' ${PLATFORM}"
+              // sh """
+              // if [ -d python-rhsm ]; then
+              //   cd python-rhsm
+              //   ../scripts/suse_build.sh 'home:kahowell' ${PLATFORM} -k \$WORKSPACE
+              //   cd ..
+              // fi
+              // """
+            }
+          }
+        }
+      }
+    }
```

Comment:
It seems that no artifacts are created. When I click on "Artifacts" link e.g. here: https://rhsm-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/blue/organizations/jenkins/subscription-manager%2Fsubscription-manager-pipeline-helper/detail/subscription-manager-pipeline-helper/1681/pipeline, then it prints that no Artifacts are available and link for following documentation: https://www.jenkins.io/doc/pipeline/tour/tests-and-artifacts/ is provided.

Reply:
That link is from the old pipeline that this code will replace.  A better link to see what this would look like is [here](https://rhsm-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/blue/organizations/jenkins/jmolet-sm-scratch/detail/PR-1/1/pipeline) .

The `publishCoverage()` and `junit()` functions will save the test results but not the raw xml files. The pipeline console output is artifacted for free without doing anything.  Adding files to artifact is easy though, what files would you like to save and for what tests?

Reply:
Thanks for the link. There is IMHO no need to save anything else ATM.